src/main/java/org/broadinstitute/dsm/route/DownloadPDFRoute.java

```diff
@@ -44,100 +46,69 @@
 
+    private static final String COVER = "cover";
+    private static final String CONSENT = "consent";
+    private static final String RELEASE = "release";
+    private static final String TISSUE = "tissue";
+    private static final String IRB = "irb";
     private static final String JSON_START_DATE = "startDate";
     private static final String JSON_END_DATE = "endDate";
 
     @Override
     public Object processRequest(Request request, Response response, String userId) throws Exception {
         logger.info(request.url());
-        if (request.url().contains(RoutePath.DOWNLOAD_PDF)) {
-            String realm = null;
-            QueryParamsMap queryParams = request.queryMap();
-            if (queryParams.value(RoutePath.REALM) != null) {
-                realm = queryParams.get(RoutePath.REALM).value();
-            }
+        QueryParamsMap queryParams = request.queryMap();
+        String realm = null;
+        if (queryParams.value(RoutePath.REALM) != null) {
+            realm = queryParams.get(RoutePath.REALM).value();
+        }
+        if (StringUtils.isNotBlank(realm)) {
             if (UserUtil.checkUserAccess(realm, userId, "pdf_download")) {
-                String requestBody = request.body();
-                if (StringUtils.isNotBlank(requestBody)) {
-                    JSONObject jsonObject = new JSONObject(requestBody);
-                    String ddpParticipantId = (String) jsonObject.get(RequestParameter.DDP_PARTICIPANT_ID);
-                    String userIdR = (String) jsonObject.get(RequestParameter.USER_ID);
-                    String configName = null;
-                    if (jsonObject.has(RequestParameter.CONFIG_NAME)) {
-                        configName = (String) jsonObject.get(RequestParameter.CONFIG_NAME);
-                    }
-                    Integer userIdRequest = Integer.parseInt(userIdR);
-                    if (!userId.equals(userIdR)) {
-                        throw new RuntimeException("User id was not equal. User Id in token " + userId + " user Id in request " + userIdR);
-                    }
-                    String medicalRecord = null;
-                    List<String> oncHistoryIDs = null;
-                    if (StringUtils.isNotBlank(ddpParticipantId)) {
-                        logger.info(request.url());
-                        String pdfType = null;
-                        if (request.url().endsWith(CONSENT_PDF)) {
-                            pdfType = CONSENT_PDF;
-                        }
-                        else if (request.url().endsWith(RELEASE_PDF)) {
-                            pdfType = RELEASE_PDF;
-                        }
-                        else if (request.url().contains(COVER_PDF)) {
-                            pdfType = COVER_PDF;
-                            medicalRecord = request.params(RequestParameter.MEDICALRECORDID);
+                if (request.url().contains(RoutePath.DOWNLOAD_PDF)) {
```

Comment:
this class in general does lots of things which is not a good idea, makes code so hard to read and debug with tons of if chaining, one of the good way would be to introduce abstraction about PDFs and separate each type of pdf creation logic in different classes under this abstraction, which would provide much more cleaner and understandable code